streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/PipelineElementFile.java

```diff
@@ -119,4 +120,14 @@ public Response getFile(
     }
     return ok(FileManager.getFile(filename));
   }
+
+  @GET
+  @Path("/allFilenames")
+  @Produces(MediaType.APPLICATION_JSON)
+  @PreAuthorize(AuthConstants.HAS_READ_FILE_PRIVILEGE)
+  public Response getAllOriginalFilenames() {
+    return ok(FileManager.getAllFiles().stream().map(fileMetadata -> fileMetadata.getOriginalFilename().toLowerCase())
+        .collect(
```

Comment:
```suggestion
        .toList();
```

Reply:
Hmmm, don't know Java now has this nice function!